sklearn/decomposition/tests/test_pca.py

```diff
@@ -474,10 +474,12 @@ def test_pca_zero_noise_variance_edge_cases(svd_solver):
     pca = PCA(n_components=p, svd_solver=svd_solver)
     pca.fit(X)
     assert pca.noise_variance_ == 0
+    # ensure no divide-by-zero error for n_components == n_features < n_samples
```

Comment:
On `main`, this is `ValueError: array must not contain infs or NaNs` error which came from a divide-by-zero warning. With that in mind, I am okay with the current phrasing.